examples/geospatial-python-urban-analysis-with-postgis/src/etl/extract/bsas_data/fetch_district.py

```
@@ -0,0 +1,9 @@
+from geopandas import GeoDataFrame as gpd
+from config import DISTRICTS_URL
+
+def fetch_districts():
+    """Fetches district data from Buenos Aires' open data API."""
+    try:
+        return gpd.read_file(DISTRICTS_URL)
+    except Exception as e:
+        raise ValueError(f"Failed to fetch districts data: {str(e)}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)